x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobManager.java

```diff
@@ -224,13 +228,26 @@ public void expandJobs(String expression, boolean allowNoMatch, ActionListener<Q
      * appropriate analysis modules/plugins.
      * The overall structure can be validated at parse time, but the exact names need to be checked separately,
      * as plugins that provide the functionality can be installed/uninstalled.
+     * If the user has not provided a categorization analyzer then set the standard one if categorization is
+     * being used at all and all the nodes in the cluster are running a version that will understand it.
      */
-    static void validateCategorizationAnalyzer(Job.Builder jobBuilder, AnalysisRegistry analysisRegistry)
-        throws IOException {
-        CategorizationAnalyzerConfig categorizationAnalyzerConfig = jobBuilder.getAnalysisConfig().getCategorizationAnalyzerConfig();
+    static void validateCategorizationAnalyzerOrSetDefault(Job.Builder jobBuilder, AnalysisRegistry analysisRegistry,
+                                                           Version minNodeVersion) throws IOException {
+        AnalysisConfig analysisConfig = jobBuilder.getAnalysisConfig();
+        CategorizationAnalyzerConfig categorizationAnalyzerConfig = analysisConfig.getCategorizationAnalyzerConfig();
         if (categorizationAnalyzerConfig != null) {
             CategorizationAnalyzer.verifyConfigBuilder(new CategorizationAnalyzerConfig.Builder(categorizationAnalyzerConfig),
                 analysisRegistry);
+        } else if (analysisConfig.getCategorizationFieldName() != null
+            && minNodeVersion.onOrAfter(MIN_NODE_VERSION_FOR_STANDARD_CATEGORIZATION_ANALYZER)) {
```

Comment:
This does seem like a good option. 

I was wonder though if it was considered to use the `Job.version` as the predicate around which default categorization to use? 

Do we ever update `Job.version`?

Ah, looking at `MlConfigMigrator`

```// Pre v5.5 (ml beta) jobs do not have a version.
        // These jobs cannot be opened, we rely on the missing version
        // to indicate this.
        // See TransportOpenJobAction.validate()
        if (jobVersion != null) {
            builder.setJobVersion(Version.CURRENT);
        }
```
I suppose that job.version is indeed mutable :/